Yosemite/Yosemite/Stores/InAppPurchaseStore.swift

```diff
@@ -275,6 +279,27 @@ private extension InAppPurchaseStore {
         return supportedCountriesCodes.contains(countryCode)
     }
 
+    /// Checks if the Site has current subscriptions via In-App Purchases
+    ///
+    func siteHasCurrentInAppPurchases(siteID: Int64) async -> Bool {
+        for await transaction in Transaction.currentEntitlements {
+            switch transaction {
+            case .verified(let transaction):
+                // If we have current entitlements, we check for its transaction token, and extract the associated siteID.
+                // If this siteID matches the current siteID, then the site has current In-App Purchases.
+                guard let token = transaction.appAccountToken,
+                      let transactionSiteID = AppAccountToken.siteIDFromToken(token),
+                      transactionSiteID == siteID else {
+                    continue
+                }
+                return true
+            default:
+                break
+            }
+        }
+        return false
+    }
```

Comment:
As mentioned in Slack, I think we should consider showing the button on sites whose IAP subscription has expired. The subscription screen will still show in the App Store, and they can resubscribe from there if they choose to.

Not needed for this PR, and happy to keep it as it is if you think the Upgrade banner is sufficient, but just a thought.

It looks like, in order to do this, we would need to iterate over `Transactions.all`, or `Transactions.latest(for: {productID}` for all the product IDs we care about here. Which may become quite clunky, especially if we add multiple-store upgrades. So this may not be the best idea, to keep the implementation simple.

Reply:
Good suggestion and good points, I've opened a follow-up issue here: https://github.com/woocommerce/woocommerce-ios/issues/10334 to investigate this further.